### ISSUE TITLE
Add TabWidthMode Property to TabStrip/TabControl

### ISSUE THREAD

**Issue by robloo** [AvaloniaUI/Avalonia#8243]
**Is your feature request related to a problem? Please describe.**

In some scenarios its common to automatically size tab item header widths to (1) fit available width and (2) keep uniform width among each tab item.

In WPF this was never supported directly; however, UWP TabView has a property:

[TabView.TabWidthMode](https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls.tabview.tabwidthmode?view=winui-2.7)
[TabViewWidthMode](https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls.tabviewwidthmode?view=winui-2.7)

![image](https://user-images.githubusercontent.com/17993847/171549973-cc9019a5-f6b9-43bd-b642-7b293e85530a.png)

This should be added to TabStrip and TabControl

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by maxkatz6** [comment]
I see possible implementation using custom tabs Panel type that arranges children in a way depending on this property.

> (1) fit available width and (2) keep uniform width among each tab item.

You can set TabControl.ItemPanel to UniformGrid with specific number of columns. Which isn't the same but can solve it mostly. Especially with converter from items count to a rows count, when you want to limit columns as well.
Together with other types of panels, including custom ones, ```TabWidthMode``` would be limiting.

Saying that, I personally don't see problems of having both TabWidthMode for simple well-known scenarios and ItemPanel for more advanced.

**Comment by robloo** [comment]
> You can set TabControl.ItemPanel to UniformGrid with specific number of columns.

Yes, this is what I'm doing now. I just realized TabView had this built in and it's pretty nice to have available.

> Saying that, I personally don't see problems of having both TabWidthMode for simple well-known scenarios and ItemPanel for more advanced.

Yes, I agree ItemPanel should still be supported.
